File: (7 kyu) Reverse words/(7 kyu) Reverse words.rs
fn reverse_words(str: &str) -> String {
    let words = str.split(" ").collect::<Vec<_>>();
    let mut res = "".to_string();
    for w in words {
        let mut x = w.to_string();
        x += " ";
        x = x.chars().rev().collect();
        res.push_str(&x);
    }
    res[1..].to_string()
}

File: (7 kyu) Reverse words/(7 kyu) Reverse words.cpp
#include <iostream>
#include <string>

using namespace std;

std::string reverse_words(std::string str)
{
    std::string res;
    std::string word;
    int length = str.length();
    for (int i = 0; i < length; i++)
    {
        if (str[i] != ' ')
        {
            word = str[i] + word;
        }
        else
        {
            if (word.length() > 0)
            {
                res += word;
                word = "";
            }
            res += " ";
        }
    }
    return res + word;
}